packages/types/src/asset-types.ts

```diff
@@ -3,218 +3,164 @@ import type {
+};
 
-interface SynthethicTrace {
-  type: "synthetic";
-  counterparty: TraceCounterpartyChain;
-  provider: string;
-}
+export type AppCurrency = KeplrAppCurrency & {
+  pegMechanism?: "collateralized" | "algorithmic" | "hybrid";
+  base?: string;
+  gasPriceStep?: GasPriceStep;
+};
 
-interface WrappedTrace {
-  type: "wrapped";
-  counterparty: TraceCounterpartyChain;
-  chain?: {
-    contract: string;
-  };
-  provider: string;
-}
+export type FeeCurrency = AppCurrency & {
+  gasPriceStep?: GasPriceStep;
+  base?: string;
+};
 
-interface BridgeTrace {
-  type: "bridge";
-  counterparty: TraceCounterpartyChain;
-  provider: string;
-}
+export type StakeCurrency = Currency & {
+  base?: string;
+};
 
-interface TraceChain {
-  channel_id: string;
-  path: string;
+export interface LogoURIs {
+  svg?: string;
+  png?: string;
 }
 
-export interface IBCTrace {
+export type IbcTransferMethod = {
+  name?: string;
   type: "ibc";
-  counterparty: TraceCounterpartyChain & {
-    channel_id: string;
-  };
-  chain: TraceChain;
-}
-
-export interface IbcCW20Trace {
-  type: "ibc-cw20";
-  counterparty: TraceCounterpartyChain & {
+  /** Counterparty chain info.
+   *  `channelId` here is commonly referred to as "destination channel". */
+  counterparty: {
+    chainName: string;
+    chainId: string;
+    sourceDenom: string;
     port: string;
-    channel_id: string;
+    channelId: string;
   };
-  chain: TraceChain & {
+  /** Osmosis chain. `channelId` here is commonly
+   *  referred to as "source channel". */
+  chain: {
     port: string;
+    channelId: string;
+    path: string;
   };
-}
+};
 
-export interface Asset {
-  denom_units: AssetDenomUnit[];
-  base: string;
+export type IntegratedBridgeTransferMethod = {
   name: string;
-  description?: string;
-  display: string;
+  type: "integrated_bridge";
+  counterparty: {
+    /** If asset is wrapped */
+    wrappedAssetId?: string;
+    unwrappedAssetId: string;
+    evmChainId: number;
+    sourceChainId: string;
+  }[];
+  wrappedAssetId?: string;
+  unwrappedAssetId?: string;
+};
+
+export type ExternalInterfaceBridgeTransferMethod = {
+  name: string;
+  type: "external_interface";
+  depositUrl?: string;
+  withdrawUrl?: string;
+};
+
+export interface Counterparty {
+  chainName: string;
+  sourceDenom: string;
+  chainType: "cosmos" | "non-cosmos" | "evm";
+  /** String keys: Cosmos. Number keys: EVM. */
+  chainId?: string | number;
   symbol: string;
+  decimals?: number;
+  logoURIs: LogoURIs;
   address?: string;
-  type_asset?: string;
-  traces: (
-    | IbcCW20Trace
-    | IBCTrace
-    | BridgeTrace
-    | WrappedTrace
-    | SynthethicTrace
-    | AdditionalMintageTrace
-    | LiquidStakeTrace
-    | TestMintageTrace
-  )[];
-  logo_URIs: LogoURIs;
-  relative_image_url: string;
-  coingecko_id?: string;
-  keywords?: string[];
-  origin_chain_name: string;
-  origin_chain_id: string;
-  price_info?: {
-    dest_coin_minimal_denom: string;
-    pool_id: string;
-  };
 }
```

Comment:
What are your thoughts on making this type an intersection based on the chain type? It would improve type safety and autocompletion.  

```ts
interface CosmosCounterparty {
  chainId: string
  ...
}

interface EVMCounterparty {
  chainId: number
  ...
}

interface NonCosmosCounterparty {
  ...
}

export type Counterparty = CosmosCounterparty | EVMCounterparty | NonCosmosCounterparty
```